lib/permissions.ts

```diff
@@ -0,0 +1,29 @@
+import { PermissionsAndroid } from "react-native";
+import CallLogs from "react-native-call-log";
+
+const checkPermission = async () => {
+  try {
+    const granted = await PermissionsAndroid.request(
+      PermissionsAndroid.PERMISSIONS.READ_CALL_LOG,
+      {
+        title: "Call Log Example",
+        message: "Access your call logs",
+        buttonNeutral: "Ask Me Later",
+        buttonNegative: "Cancel",
+        buttonPositive: "OK",
+      },
+    );
+    if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+      console.log(CallLogs);
+      CallLogs.load(-1).then((c) => console.log(c));
```

Comment:
this is where the filtering based on array of phone numbers should be happening

Reply:
```js
/* List call logs matching the filter */
const filter = {
  minTimestamp: 1571835032,    // (Number or String) timestamp in milliseconds since UNIX epoch
                               // if this filter is set, load(limit, filter) will only return call logs with timestamp >= minTimestamp

  maxTimestamp: 1571835033,    // (Number or String) timestamp in milliseconds since UNIX epoch
                               //
                               // if this filter is set, load(limit, filter) will only return call logs with timestamp <= maxTimestamp

  phoneNumbers: '+1234567890', // (String or an Array of String)
                               // if this filter is set, load(limit, filter) will only return call logs for this/these phone numbers

  types: 'MISSED',             // (Single value / Array of Strings / CallType enums)
                               // if this filter is set, load(limit, filter) will only return call logs of a specific type
}

const callLogs = await CallLogs.load(-1, filter) // applies filter with no limit (also works with a limit)
```

Reply:
```
`    if (granted === PermissionsAndroid.RESULTS.GRANTED) {
      const contactNumbers = await getContactNumbers();
      const filter = {
        phoneNumbers: contactNumbers,
      };

      // Load call logs using the filter
      CallLogs.load(-1, filter).then((callLogs) => console.log(callLogs));
    } else {
      console.log("Call Log permission denied");
    }
  } catch (error) {
    console.error("Error checking call log permission:", error);
  }
};
`
```
getting contact numbers:
```
`
const getContactNumbers = async () => {
  try {
    const response = await fetch(
      "https://worker-turso-ts.technoculture.workers.dev/tasks",
    );
    const tasks = await response.json();
    console.log("Tasks fetched successfully", tasks);

    const contactNumbers = tasks.map((task: any) => task.contactNumber);
    return contactNumbers;
  } catch (error) {
    console.error("Error fetching contact numbers:", error);
    return [];
  }
};
`

```